src/AbstractOperations/computations.jl

```diff
@@ -0,0 +1,61 @@
+struct Computation{R, T, O, G}
+      operation :: O
+         result :: R
+           grid :: G
+    return_type :: T
+end
+
+Computation(op, result; return_type=Array) = Computation(op, result, op.grid, return_type)
+
+architecture(comp::Computation) = architecture(comp.result)
+Base.parent(comp::Computation) = comp
```

Comment:
This is because we call `parent` on arguments to `HorizontalAverage`. We can take the horizontal average of 1. a field, or 2. a computation.

It is a bit confusing otherwise, because computation doesn't behave like an array in any other way (though we could write some functions that, in fact, achieve this behavior).